c_wrapper/lua-mongoc-collection.c

```diff
@@ -631,13 +631,13 @@ lua_mongo_collection_destroy(lua_State *L)
 }
 
 
-
 int
 lua_mongo_collection_create_index(lua_State *L)
 {
     bool throw_error = false;
     bool created = false;
 
+    char *idxName = "";
```

Comment:
You would probably just do `char *idxName;` here.
With other variables, snake case is the common style in C, so it would now be `char *index_name;`

However, that is all irrelevant. You don't need to use asprintf